lib/puppet/pops/types/types.rb

```diff
@@ -2799,6 +2800,12 @@ def really_instance?(o, guard = nil)
     resolved_type.really_instance?(o, guard)
   end
 
+  # @return `nil` to prevent serialization of the type_expr used when first initializing this instance
+  # @api private
+  def type_expr
+    nil
```

Comment:
It would, at least partly. I've created a generic Object reader/writer that makes the assumption that the arguments passed to the initializer are accessible as readers. In this case the initializer takes three arguments, the last one (the resolved expression) is optional, but only if the second argument (the unresolved expression) is given instead. Conversely, the second argument is of course not needed when the third argument is given. It must be passed in as `nil` though, since unnamed parameters are used.